packages/utils/src/types/errors.ts

```diff
@@ -0,0 +1,19 @@
+import { ErrorResponse } from 'ontime-types';
+
+export function isErrorResponse(error: unknown): error is ErrorResponse {
+  return (
+    typeof error === 'object' &&
+    error !== null &&
+    'message' in error &&
+    typeof (error as Record<string, unknown>).message === 'string'
+  );
+}
+
+export function toErrorResponse(maybeError: unknown): ErrorResponse {
+  if (isErrorResponse(maybeError)) return maybeError;
+  try {
+    return new Error(JSON.stringify(maybeError));
+  } catch {
+    return new Error(String(maybeError));
+  }
+}
```

Comment:
We start accumulating several of these, from top of my head `maybeAxiosError`, `maybeError`, `unpackError`

I think it is perfectly fine to narrow it down to the context, so, `maybeAxiosError` only makes sense in the client and here we likely just want to handle generic errors of Error.
We dont have a complex use case and I wonder if the `unpackError` function could fit the bill here. Maybe it should be called `getErrorMessage()`

For reference, this is also a good article about it, with a resolution very similar to yours.
In the article we can find the motivation for the different bits of logic so that we can see how much of it applies to our codebase
https://kentcdodds.com/blog/get-a-catch-block-error-message-with-typescript

Reply:
> For reference, this is also a good article about it, with a resolution very similar to yours

That is because is is lifted straight from there 😉

Reply:
hahaha, nice...
In this case I still prefer the original offered by Kent C Dodds
```ts
function getErrorMessage(error: unknown) {
  if (error instanceof Error) return error.message
  return String(error)
}
```

I dont think we have to worry about the edge cases that the more complex solution handles